core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -247,10 +248,16 @@ public SystemSessionProperties(
                         false,
                         false),
                 integerProperty(
-                        TASK_WRITER_COUNT,
-                        "Default number of local parallel table writer jobs per worker",
-                        taskManagerConfig.getWriterCount(),
-                        value -> validateValueIsPowerOfTwo(value, TASK_WRITER_COUNT),
+                        UNPARTITIONED_TASK_WRITER_COUNT,
+                        "Number of writers per task when data is unpartitioned and prefer partitioning is disabled",
+                        taskManagerConfig.getUnpartitionedWriterCount(),
+                        value -> validateValueIsPowerOfTwo(value, UNPARTITIONED_TASK_WRITER_COUNT),
+                        false),
+                integerProperty(
+                        PARTITIONED_TASK_WRITER_COUNT,
+                        "Number of writers per task when data is partitioned and prefer partitioning is enabled",
```

Comment:
`prefer partitioning is enabled` -> it might be enabled, but engine might still skip it.

Maybe:
```
Number of writers per task when prefer partitioning is used
```